build_tools/azure/test_script.sh

```diff
@@ -29,6 +29,18 @@ if [[ "$BUILD_REASON" == "Schedule" ]]; then
     export SKLEARN_RUN_FLOAT32_TESTS=1
 fi
 
+# debug
+echo full env
+env
+
+echo env matching build
+env | grep -i build
+
+if [[ "$BUILD_SOURCEVERSIONMESSAGE" =~ \[float32\ test\] ]]; then
```

Comment:
I do not think this works for getting the commit message on PRs. We likely re-use: https://github.com/scikit-learn/scikit-learn/blob/main/build_tools/azure/get_commit_message.py to set the variable in Azure.